DuckDuckGo/OmniBar.swift

```diff
@@ -156,26 +156,32 @@ class OmniBar: UIView {
         if let query = appUrls.searchQuery(fromUrl: url) {
             textField.text = query
         } else {
-            textField.attributedText = demphasisePath(forUrl: url)
+            textField.attributedText = OmniBar.demphasisePath(forUrl: url)
         }
     }
 
-    private func demphasisePath(forUrl url: URL) -> NSAttributedString {
-        let s = url.absoluteString
-        let parts = s.split(separator: "/")
-        let offset = (parts[0] + "//" + parts[1]).count
+    public class func demphasisePath(forUrl url: URL) -> NSAttributedString? {
         
+        let s = url.absoluteString
         let attributedString = NSMutableAttributedString(string: s)
-        let theme = ThemeManager.shared.currentTheme
-        
-        let idx = s.unicodeScalars.index(s.unicodeScalars.startIndex, offsetBy: offset)
```

Comment:
This is the line causing the crash.  When we removed puny decoding this should have been changed then anyway as it's redundant if the string is always ascii. However, the URL components approach is better anyway.